### PR TITLE
Fixed and renovated Rust example 08 grpc.

### PR DESCRIPTION
**Pull request by marvin-hansen**
Fixed and renovaeted rust-examples/08-grpc-client-server/
Resolves: https://github.com/bazelbuild/examples/pull/489#issuecomment-2321617323

The CI failure of the example was caused  by a bizarre issue that only occurs on linux, as documented https://github.com/bazelbuild/rules_rust/issues/2806

I've applied the known fix. Furthermore, the cause of the issue seems to be a change in the format of how aliases 
are generated for downloaded dependencies referenced in the example. To prevent this from ever happening again, I've vendored all dependencies and updated the Bazel build files accordingly.

### CODE REVIEW

**Comment by marvin-hansen** [comment]
Friendly ping.
Build is green so this should be easy to review.

**Review by Wyverald — COMMENTED** [review]
hmm. would you say this is an intended way to use rules_rust (vendoring BUILD files)? Do you think we should add a rules_rust maintainer to review this too?

**Comment by marvin-hansen** [comment]
Well,

Adding a rules_rust maintainer is an excellent idea.

While I've ported some of the vendoring examples to Bazelmod and
contributed it to the rules a while ago, i always got puzzled why
rules_rust actually has  three different vendoring modes so it's perfectly
possible I missed something.

**Review by illicitonion — APPROVED** [review]
LGTM, thanks!

I'd say it's equally idiomatic to:
1. Vendor the deps
2. Pull in the deps via a Cargo.toml file and `crate.from_cargo` calls
3. Pull the deps in via `crate.spec` calls

There are different trade-offs to each, and those are covered by examples 05, 06, and 07, so it doesn't hugely matter which we use in this example - vendoring seems as good as any :)

**Comment by marvin-hansen** [comment]
Sounds good, thanks!

Yes, the leftover file can be removed.
 Good catch.

Will add a cleanup commit relatively soon.

Thanks the prompt review.

File: rust-examples/08-grpc-client-server/Cargo.toml [hunk 1]
```diff
@@ -18,10 +18,10 @@ readme = "README.md"
 # Internal crates
 proto_bindings = { path = "proto_bindings" }
 # External crates
-prost = { version = "0.12.6" }
-prost-types = { version = "0.12.6", default-features = false }
-tonic = { version = "0.11.0", features = ["transport"] }
-tonic-build = "0.11.0"
+prost = { version = "0.13.2" }
```
[the remark this answers is not in the document]
Comment (marvin-hansen) [reply rust-examples/08-grpc-client-server/Cargo.toml]: ok, the Cargo files are all deleted now. CI is running again.

**Comment by marvin-hansen** [comment]
@illicitonion Thank you for the elaboration.

I guess it all makes sense and at some point when I find the time, I have to put this into the vendoring documentation so people know the trade-offs upfront.  

In summary:

* from_cargo - Requires a Cargo.tom; Need to add a Bazelmod example one day.
A)  mode = "local"
B)   mode = "remote"

* from_spec - There is an example, too; Check 
A)  mode = "local",
B)   mode = "remote" 
 
Local means Bazel downloads each crate and stores the entire crate in the repo under the thirdparty / crate folder

Remote means Bazel only stores the BUILD.bazel files and presumably puts the downloaded crates in the cache 

I would say, for the Rust examples remote is fine as it does not pollute the repo. 

However, I've put a note in the readme explaining the different vendor modes and how to switch mode.

**Review by Wyverald — APPROVED** [review]

**Pull request merged**